docs/apis/subsystems/muc/index.md

```diff
@@ -0,0 +1,310 @@
+All you need to do is add a definition for your cache and you are ready to start working with the Cache API.
+
+### Creating a definition
+
+Cache definitions exist within the db/caches.php file for a component/plugin.<br />
+In the case of core that is the moodle/lib/db/caches.php file, in the case of a module that would be moodle/mod/myplugin/db/caches.php.
+
+The definition is used API in order to understand a little about the cache and what it is being used for, it also allows the administrator to set things up especially for the definition if they want.
+From a development point of view the definition allows you to tell the API about your cache, what it requires, and any (if any) advanced features you want it to have.<br />
+The following shows a basic definition containing just the bare minimum:
+
+```php
+// moodle/mod/myplugin/db/caches.php
+$definitions = [
+    'somedata' => [
+        'mode' => cache_store::MODE_APPLICATION,
+    ]
+];
+```
+
+This informs the API that the myplugin module has a cache called *somedata* and that it is an application (globally shared) cache.
+
+When creating a definition thats the bare minimum, to provide an area *(somedata)* and declare the type of the cache application, session, or request.
+
+- An application cache is a shared cache, all users can access it.
+- Session caches are scoped to a single users session, but may not actually be stored in the session.
+- Request caches you can think of as static caches, only available to the user owning the request, and only alive until the end of the request.
+There are of course many more options available that allow you to really take the cache by the reigns, you can read about some of the important ones further on, or skip ahead to [#The definition](https://docs.moodle.org/dev/#The_definition) section which details the available options in full.
+
+Please note that for each definition a language string with the name *cachedef_* followed by the name of the definition is expected. Using the example above you would have to define:
+
+```php
+// moodle/mod/myplugin/lang/en/mod_myplugin.php
+$string['cachedef_somedata'] = 'This is the description of the cache somedata';
+```
+
+### Getting a cache object
+
+Once your definition has been created you should bump the version number so that Moodle upgrades and processes the definitions file at which point your definition will be useable.
+
+Now within code you can get a cache object corresponding to the definition created earlier.
+
+```php
+$cache = cache::make('mod_myplugin', 'somedata');
+```
+
+The cache::make method is a factory method, it will create a cache object to allow you to work with your cache. The cache object will be one of several classes chosen by the API based upon what your definition contains. All of these classes will extend the base cache class, and in nearly all cases you will get one of cache_application, cache_session, or cache_request depending upon the mode you selected.
```

Comment:
```suggestion
The `cache::make()` method is a factory method, it will create a cache object to allow you to work with your cache. The cache object will be one of several classes chosen by the API based upon what your definition contains. All of these classes will extend the base cache class, and in nearly all cases you will get one of `cache_application`, `cache_session`, or `cache_request` depending upon the mode you selected.
```